### PR TITLE
Add mixer subscription update script

### PR DESCRIPTION
**Pull request by ChristopherBiscardi**

### CODE REVIEW

File: functions/update-mixer-subscriptions/update-mixer-subscriptions.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
+// TODO Chris: Remove this imp;ort
+const AppSecrets = require('../../secrets');
```
Comment (ChristopherBiscardi) [inline functions/update-mixer-subscriptions/update-mixer-subscriptions.js]:
```suggestion
```

File: functions/update-mixer-subscriptions/update-mixer-subscriptions.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
+// TODO Chris: Remove this imp;ort
```
Comment (ChristopherBiscardi) [inline functions/update-mixer-subscriptions/update-mixer-subscriptions.js]:
```suggestion
```

File: functions/update-mixer-subscriptions/update-mixer-subscriptions.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
+const fetch = require('node-fetch');
+
+const Mixer = require('@mixer/client-node');
+const client = new Mixer.Client(new Mixer.DefaultRequestRunner());
+
+client.use(
+  new Mixer.OAuthProvider(client, {
+    // TODO Chris: Swap these local variables with env variables
+    clientId: AppSecrets.PARTY_CORGI_CLIENT_ID,
+    secret: AppSecrets.PARTY_CORGI_SECRET
```
Comment (ChristopherBiscardi) [inline functions/update-mixer-subscriptions/update-mixer-subscriptions.js]:
```suggestion
    secret: process.env.PARTY_CORGI_SECRET
```

File: functions/update-mixer-subscriptions/update-mixer-subscriptions.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
+const fetch = require('node-fetch');
+
+const Mixer = require('@mixer/client-node');
+const client = new Mixer.Client(new Mixer.DefaultRequestRunner());
+
+client.use(
+  new Mixer.OAuthProvider(client, {
+    // TODO Chris: Swap these local variables with env variables
+    clientId: AppSecrets.PARTY_CORGI_CLIENT_ID,
```
Comment (ChristopherBiscardi) [inline functions/update-mixer-subscriptions/update-mixer-subscriptions.js]:
```suggestion
    clientId: process.env.PARTY_CORGI_CLIENT_ID,
```

File: functions/update-mixer-subscriptions/update-mixer-subscriptions.js [hunk 5]
```diff
@@ -0,0 +1,123 @@
+const fetch = require('node-fetch');
+
+const Mixer = require('@mixer/client-node');
+const client = new Mixer.Client(new Mixer.DefaultRequestRunner());
+
+client.use(
+  new Mixer.OAuthProvider(client, {
+    // TODO Chris: Swap these local variables with env variables
```
Comment (ChristopherBiscardi) [inline functions/update-mixer-subscriptions/update-mixer-subscriptions.js]:
```suggestion
```

File: functions/update-mixer-subscriptions/update-mixer-subscriptions.js [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+    secret: AppSecrets.PARTY_CORGI_SECRET
+  })
+);
+
+exports.handler = async (event, context) => {
+  try {
+    // 1. Fetch list from /fetch-mixer-names
+    const streamerList = await fetch(
+      // TODO Chris: Swap this local variable with an env variable
+      `${AppSecrets.SITE_URL}/.netlify/functions/fetch-mixer-names`,
```
Comment (ChristopherBiscardi) [inline functions/update-mixer-subscriptions/update-mixer-subscriptions.js]:
```suggestion
      `${process.env.SITE_URL}/.netlify/functions/fetch-mixer-names`,
```

File: functions/update-mixer-subscriptions/update-mixer-subscriptions.js [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+    clientId: AppSecrets.PARTY_CORGI_CLIENT_ID,
+    secret: AppSecrets.PARTY_CORGI_SECRET
+  })
+);
+
+exports.handler = async (event, context) => {
+  try {
+    // 1. Fetch list from /fetch-mixer-names
+    const streamerList = await fetch(
+      // TODO Chris: Swap this local variable with an env variable
```
Comment (ChristopherBiscardi) [inline functions/update-mixer-subscriptions/update-mixer-subscriptions.js]:
```suggestion
```

File: functions/update-mixer-subscriptions/update-mixer-subscriptions.js [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+      .then(data => data)
+      .catch(err => console.log(err));
+
+    const mixerUrls = streamerList.mixer;
+
+    // 2. Get current hooks from Mixer
+    const currentSubscriptions = await client
+      .request('GET', `/hooks`, {
+        headers: {
+          // TODO Chris: Swap this local variable with an env variable
```
Comment (ChristopherBiscardi) [inline functions/update-mixer-subscriptions/update-mixer-subscriptions.js]:
```suggestion
```

File: functions/update-mixer-subscriptions/update-mixer-subscriptions.js [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+          Authorization: `Secret ${AppSecrets.PARTY_CORGI_SECRET}`
```
Comment (ChristopherBiscardi) [inline functions/update-mixer-subscriptions/update-mixer-subscriptions.js]:
```suggestion
          Authorization: `Secret ${process.env.PARTY_CORGI_SECRET}`
```

File: functions/update-mixer-subscriptions/update-mixer-subscriptions.js [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+        .request(`POST`, `/hooks`, {
+          body: {
+            events: [`channel:${id}:broadcast`],
+            kind: 'web',
+            // TODO Chris: This URL should hit the deployed endpoint for the
+            // catch-mixer-events function
+            url: ``
+          },
+          headers: {
+            // TODO Chris: Swap these local variables with env variables
```
Comment (ChristopherBiscardi) [inline functions/update-mixer-subscriptions/update-mixer-subscriptions.js]:
```suggestion
```

File: functions/update-mixer-subscriptions/update-mixer-subscriptions.js [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+
+    idsToAdd.map(id => {
+      client
+        .request(`POST`, `/hooks`, {
+          body: {
+            events: [`channel:${id}:broadcast`],
+            kind: 'web',
+            // TODO Chris: This URL should hit the deployed endpoint for the
+            // catch-mixer-events function
+            url: ``
```
Comment (ryancharris) [inline functions/update-mixer-subscriptions/update-mixer-subscriptions.js:99]: @ChristopherBiscardi Can you add the URL here and remove the comment? Thanks 🙌

File: functions/update-mixer-subscriptions/update-mixer-subscriptions.js [hunk 12]
```diff
@@ -0,0 +1,119 @@
...
+          },
+          headers: {
+            Authorization: `Secret ${AppSecrets.PARTY_CORGI_SECRET}`,
```
Comment (ryancharris) [inline functions/update-mixer-subscriptions/update-mixer-subscriptions.js]: I need to change `AppSecret` to `process.env` on this line and the one below

**Comment by ryancharris** [comment]
@ChristopherBiscardi Can you hit the note above and check this out/merge it? 🙏

**Pull request merged**